nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,272 @@
+        '';
+      };
+
+      log-level = mkOption {
+        type = types.nullOr types.str;
+        default = "debug";
+        description = ''
+          The log level to use [ "error", "warn", "info", "debug" ]. [TS_PROM_LOG_LEVEL] (default "debug")
+        '';
+      };
+
+      metrics-cache-size = mkOption {
+        type = types.nullOr types.int;
+        default = 10000;
+        description = ''
+          maximum number of metric names to cache [TS_PROM_METRICS_CACHE_SIZE] (default 10000)
+        '';
+      };
+
+      migrate = mkOption {
+        type = types.str;
+        default = "true";
+        example = "false";
+        description = ''
+          Update the Prometheus SQL to the latest version [TS_PROM_MIGRATE] (default true)
+        '';
+      };
+
+      scheduled-election-interval = mkOption {
+        type = types.nullOr types.str;
+        default = "5s";
+        description = ''
+          Interval at which scheduled election runs. This is used to select a leader and confirm that we still holding the advisory lock. [TS_PROM_SCHEDULED_ELECTION_INTERVAL] (default 5s)
+        '';
+      };
+
+      tput-report = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          interval in seconds at which throughput should be reported [TS_PROM_TPUT_REPORT]
+        '';
+      };
+
+      web-listen-address = mkOption {
+        type = types.str;
+        default = ":9201";
+        description = ''
+          Address to listen on for web endpoints. [TS_PROM_WEB_LISTEN_ADDRESS] (default ":9201")
+        '';
+      };
+
+      web-telemetry-path = mkOption {
+        type = types.str;
+        default = "/metrics";
+        description = ''
+          Address to listen on for web endpoints. [TS_PROM_WEB_TELEMETRY_PATH] (default "/metrics")
+        '';
+      };
+
+      extraOptions = mkOption {
+        default = [];
+        type = types.listOf types.str;
+        example = literalExample ''
+          [ "-async-acks"
+          ]
+        '';
+        description = ''
+          Extra command line arguments to pass to timescale-prometheus.
+        '';
+      };
+
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = (cfg.db-host != "" && cfg.db-host != "localhost") -> (cfg.db-passwordFile != null);
+        message = "When services.timescale-prometheus.db-host is changed from default, services.timescale-prometheus.db-passwordFile must be set!";
+      }
+    ];
+
+    meta.maintainers = with lib.maintainers; [ _0x4A6F ];
+
+    systemd.services.timescale-prometheus = {
+      description = "An open-source analytical platform for Prometheus metrics";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
+      after = [ "postgresql.service" ];
+
+      environment = {
+        TS_PROM_ASYNC_ACKS = "${toString cfg.async-acks}";
```

Comment:
I could be entirely wrong here but I assume values under `environment` automatically cast to string anyways. Is that not the case?

Additionally quoting is not required:
```
TS_PROM_ASYNC_ACKS = toString cfg.async-acks;

# or if toString is not required
TS_PROM_ASYNC_ACKS = cfg.async-acks;
```

Reply:
Now I know. Thanks.